app/models/course/assessment/answer.rb

```diff
@@ -35,9 +35,13 @@ class Course::Assessment::Answer < ActiveRecord::Base
   def auto_grade!
     fail ArgumentError unless question.auto_gradable?
 
-    save!
-    create_auto_grading!
-    Course::Assessment::Answer::AutoGradingJob.new(auto_grading)
+    self.class.transaction do
+      save!
```

Comment:
Yes, exactly. This is not supposed to fail. If it fails it's something beyond our control IMO. Hence my use of the bang in the function name.